doc/qos/SONiC_ROCEv2_support.md

```diff
@@ -0,0 +1,488 @@
+
+### 1.2.2 Container
+The front end code change will be done in management-framework container including:
+- XML file for the CLI
+- Python script to handle CLI request (actioner)
+- Jinja template to render CLI output (renderer)
+- OpenConfig YANG model for QoS buffer openconfig-qos.yang
+- SONiC QOS buffer model on Redis DB schema
+- transformer functions to
+   * convert OpenConfig YANG model to SONiC YANG model for buffer related configurations
+
+### 1.2.3 SAI Overview
+
+# 2 Functionality
+
+## 2.1 Target Deployment Use Cases
+Manage/configure QoS buffer configurations via gNMI, REST and CLI interfaces
+
+## 2.2 Functional Description
+Provide CLI, gNMI and REST support for QoS buffer onfigurations.
+
+## 2.3 Backend change to support new configurations
+Provide change in management framework and buffermgrd modules.
+
+## 2.4 Configurations already supported in QoS for TC mapping
+
+## 2.4.1 dot1p & DSCP to Traffic Class mapping: -
+DSCP is preferred over dot1p when both are present in the data traffic but both is not expected to co-exist.
+```
+sonic(config)# qos map dot1p-tc rocev2_dot1p-to-tc
+sonic(conf-dot1p-tc-map-rocev2_dot1p-to-tc)# dot1p 4 traffic-class 4
+sonic(conf-dot1p-tc-map-rocev2_dot1p-to-tc)#
+
+sonic(config)# qos map dscp-tc rocev2_dscp-to-tc
+sonic(conf-dscp-tc-map-rocev2_dscp-to-tc)# dscp 6 traffic-class 4
+sonic(conf-dscp-tc-map-rocev2_dscp-to-tc)#
+```
+### 2.4.2 Traffic class to dot1p & DSCP mapping:
+```
+sonic(config)# qos map tc-dot1p rocev2_tc-to-dot1p
+sonic(conf-tc-dot1p-map-rocev2_tc-to-dot1p)# traffic-class 4 dot1p 4
+sonic(conf-tc-dot1p-map-rocev2_tc-to-dot1p)#
+
+sonic(config)# qos map tc-dscp rocev2_tc-to-dscp
+sonic(conf-tc-dscp-map-rocev2_tc-to-dscp)# traffic-class 4 dscp 6
+sonic(conf-tc-dscp-map-rocev2_tc-to-dscp)#
+```
+## 2.4.3 Traffic Class to Queue mapping:
+```
+sonic(config)# qos map tc-queue rocev2_tc-q
+sonic(conf-tc-queue-map-rocev2_tc-q)# traffic-class 4 queue 4
+```
+### 2.4.4 Traffic Class to Queue mapping: (PFC Pause frames priority only 3 & 4 are supported HW)
+
+```
+sonic(config)# qos map tc-pg rocev2_tc-pg
+sonic(conf-tc-pg-map-rocev2_tc-pg)# traffic-class 4 priority-group 4
+```
+
+### 2.4.5 ECN configuration:
+```
+sonic(config)# qos wred-policy rocev2_wred
+sonic(conf-wred-rocev2_wred)# green minimum-threshold 1048 maximum-threshold 2097 drop-probability 5
+sonic(conf-wred-rocev2_wred)# ecn all
+
+PFC watchdog configuration: (PFC pause frame storm control)
+---------------------------
+sonic(config)# priority-flow-control watchdog polling-interval 300
+sonic(config)# priority-flow-control watchdog counter-poll
+```
+# 3 Design
+
+## 3.1 Overview
+
+Enhancing the management framework backend code and transformer methods to add support for QoS buffer.
+
+## 3.2 DB Changes
+
+### 3.2.1 CONFIG DB
+Existing QoS buffer tables will be used and a new field will be introduced in PORT table to avoid taking the buffer-mgr created profile based

```

Comment:
Is the PORT Table schema(with the new field) captured in this doc?

Reply:
will update.